src/algebra/homology/chain_complex.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.graded_object
+import category_theory.differential_object
+
+/-!
+# Chain complexes
+
+We define a chain complex in `V` as a differential `ℤ`-graded object in `V`.
+
+This is fancy language for the obvious definition,
+and it seems we can use it straightforwardly:
+
+```
+example (C : chain_complex V) : C.X 5 ⟶ C.X 6 := C.d 5
+```
+
+We define the forgetful functor to `ℤ`-graded objects, and show that
+`chain_complex V` is concrete when `V` is, and `V` has coproducts.
+-/
+
+universes v u
+
+open category_theory
+open category_theory.limits
+
+section
+variables (V : Type u) [𝒱 : category.{v} V]
+include 𝒱
+
+variables [has_zero_morphisms.{v} V]
+
+/--
+A chain complex in `V` is "just" a differential `ℤ`-graded object in `V`.
+-/
+-- Actually, because I chose the "shift" to correspond to the +1 direction,
+-- this is what would usually be called a cochain complex.
+-- We should definitely bikeshed this. :-)
```

Comment:
I guess I was assuming that I would take these out during review. Perhaps I should have made a comment here on GitHub, rather than making them in the code.

On the other hand, the more documentation the better...